File: time_exeded_solutions/multiples/time_exeded_multiples.py
# print result mod 10^9+7
P = pow(10, 9) + 7

x, y = map(int, input().split())
n = int(input())

# // is for integer division
# n//x is the number of how many multiples of x in [0..n]
# calculate the sum using the summation formula n*(n+1)/2
# then multiplay it by x


def calc_multiples(x, n):
    s = 0
    for i in range(0, n+1, x):
        s = (s + i) % P
    return s


# multiples of x
mx = calc_multiples(x, n)
# multiples of y
my = calc_multiples(y, n)
# multiples of x*y (the common multiples)
mxy = calc_multiples(x*y, n)

# the result is the sum of x and y multiples - the common multiples
s = (mx + my - mxy) % P
print(s)

File: time_exeded_solutions/multiples/time_exeded_multiples.cpp
#include <iostream>

using namespace std;

#define P 1000000007;

int calc_multiples(int x, int n)
{
    int sum = 0;
    for (int i = 0; i <= n; i += x)
    {
        sum = (sum + i) % P;
    }
    return sum;
}

int main(int argc, char const *argv[])
{
    int x, y, n, s;
    int mx, my, mxy;
    cin >> x >> y;
    cin >> n;

    mx = calc_multiples(x, n);
    my = calc_multiples(y, n);
    mxy = calc_multiples(x * y, n);

    s = (mx + my - mxy) % P;
    cout << s << endl;
    return 0;
}
